common4j/src/main/com/microsoft/identity/common/java/providers/oauth2/AuthorizationRequest.java

```diff
@@ -241,6 +258,11 @@ public URI getAuthorizationRequestAsHttpRequest() throws ClientException {
             final CommonURIBuilder builder = new CommonURIBuilder(getAuthorizationEndpoint());
             builder.addParametersIfAbsent(ObjectMapper.serializeObjectHashMap(this));
             builder.addParametersIfAbsent(mExtraQueryParams);
+
+            if (!StringUtil.isNullOrEmpty(mState)) {
+                builder.setParameter("state", getEncodedState()); // pass the URL safe state to the URL builder.
+            }
```

Comment:
Were we not even attaching state previously?

Reply:
We were, but it was only URL encoded. We need to Base64 encode it when building the URL.

Reply:
Was that happening in a different place in the code? Can you share the existing code where that was happening?

Reply:
I'm asking because I'm not seeing any code replaced in this PR that was doing the following:

```java
builder.setParameter("state", mState);
```

Reply:
It seems like what we're doing here is first attaching the only URL encoded state to the builder and then later in the flow we're patching (overriding) it with Base64 encoded variant. I think this would make debugging harder. Probably a better idea to set the state only once and correctly in the first place on the authorization request (builder).

Reply:
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1750/commits/49bf80b27b2e2c5a3e93047f563df5603050beb9